public/app/plugins/datasource/loki/query_utils.test.ts

```diff
@@ -61,3 +62,39 @@ describe('getHighlighterExpressionsFromQuery', () => {
     expect(getHighlighterExpressionsFromQuery('{foo="bar"} |~ `\\w+`')).toEqual(['\\w+']);
   });
 });
+
+describe('getNormalizedLokiQuery', () => {
+  function expectNormalized(inputProps: Object, outputQueryType: LokiQueryType) {
+    const input: LokiQuery = { refId: 'A', expr: 'test1', ...inputProps };
+    const output = getNormalizedLokiQuery(input);
+    expect(output).toStrictEqual({ refId: 'A', expr: 'test1', queryType: outputQueryType });
+  }
+
+  it('handles no props case', () => {
+    expectNormalized({}, LokiQueryType.Range);
+  });
+
+  it('handles old-style instant case', () => {
+    expectNormalized({ instant: true, range: false }, LokiQueryType.Instant);
+  });
+
+  it('handles old-style range case', () => {
+    expectNormalized({ instant: false, range: true }, LokiQueryType.Range);
+  });
+
+  it('handles new+old style instant', () => {
+    expectNormalized({ instant: true, range: false, queryType: LokiQueryType.Range }, LokiQueryType.Range);
+  });
+
+  it('handles new+old style range', () => {
+    expectNormalized({ instant: false, range: true, queryType: LokiQueryType.Instant }, LokiQueryType.Instant);
+  });
+
+  it('handles new<>old conflict (new wins), range', () => {
```

Comment:
in this approach, if new-style and old-style prop both exist, and say different things, new-style prop wins. i think this is the correct thing to do, but not 100% sure.